### PR TITLE
Test Kubernetes 1.15 in CI

### PR DESCRIPTION
**Pull request by howardjohn**
Will enable in postsubmit once/if this passes

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: tests/integration/pilot/main_test.go [hunk 1]
```diff
@@ -37,18 +38,30 @@ var (
...
 // If a test requires a custom install it should go into its own package, otherwise it should go
 // here to reuse a single install across tests.
 func TestMain(m *testing.M) {
 	framework.
 		NewSuite(m).
 		Setup(func(ctx resource.Context) (err error) {
-			crd, err := ioutil.ReadFile("testdata/service-apis-crd.yaml")
-			if err != nil {
-				return err
+			if supportsCRDv1(ctx) {
```
Comment (mandarjog) [inline tests/integration/pilot/main_test.go:57]: Nit Nesting. If !supportsCRDv1() return nil

File: tests/integration/pilot/main_test.go [hunk 2]
```diff
@@ -37,18 +38,30 @@ var (
 	apps = &common.EchoDeployments{}
 )
 
+func supportsCRDv1(ctx resource.Context) bool {
+	ver, err := ctx.Clusters()[0].GetKubernetesVersion()
+	if err != nil {
+		return true
+	}
+	serverVersion := fmt.Sprintf("%s.%s", ver.Major, ver.Minor)
+	return serverVersion >= "1.16"
```
Comment (elfinhe) [inline tests/integration/pilot/main_test.go:47]: => `ver.Major == 1 && ver.Minor >= 16`?

**Review by elfinhe — APPROVED** [review]

File: pkg/test/framework/components/echo/kube/deployment.go [hunk 3]
```diff
@@ -378,6 +380,15 @@ const DefaultVMImage = "app_sidecar_ubuntu_bionic"
...
 	ctx resource.Context, cfg echo.Config,
 	settings *image.Settings, cluster resource.Cluster) (serviceYAML string, deploymentYAML string, err error) {
+	ver, err := cluster.GetKubernetesVersion()
+	if err != nil {
+		return "", "", err
+	}
+	supportStartupProbe := true
+	if ver.Minor < "16" {
+		// Added in Kubernetes 1.16
+		supportStartupProbe = false
```
Comment (nmittler) [inline pkg/test/framework/components/echo/kube/deployment.go:390]: was it on by default in 1.16?

**Review by nmittler — APPROVED** [review]

File: pkg/test/framework/components/echo/kube/deployment.go (hunk 3 above)
Comment (howardjohn) [reply pkg/test/framework/components/echo/kube/deployment.go:390]: I think so? certainly in kind it is

**Pull request merged**